app/frontend/src/components/ches/UploadContexts.vue

```diff
@@ -0,0 +1,385 @@
+<template>
+  <div>
+    <div v-if="excelParsed">
+      <v-simple-table dense class="contexts-table">
+        <template v-slot:default>
+          <thead>
+            <th
+              v-for="header in excel.headers[0]"
+              :key="header"
+              class="text-left"
+            >
+              {{ header }}
+            </th>
+          </thead>
+          <tbody>
+            <tr v-for="row in excel.data" :key="excel.data.indexOf(row)">
+              <td v-for="val in row" :key="val" class="text-left truncate pl-0">
+                {{ val }}
+              </td>
+            </tr>
+          </tbody>
+        </template>
+      </v-simple-table>
+    </div>
+
+    <div
+      v-if="!excelParsed"
+      ref="dropZone"
+      @drop.prevent="addExcelFile($event)"
+      @click="fileInputBrowse()"
+      @dragover.prevent="dragover = true"
+      @dragenter.prevent="dragover = true"
+      @dragleave.prevent="dragover = false"
+      :class="['drop-zone', { 'grey lighten-2': dragover }]"
+    >
+      <v-container>
+        <v-row
+          class="d-flex flex-column"
+          dense
+          align="center"
+          justify="center"
+        >
+          <p class="pt-5 mb-1">
+            Drag &amp; Drop your Excel file here, or click to Browse.
+          </p>
+          <p class="">
+            A sample Excel file containing contexts is available to download on the <router-link :to="{ name: 'MergeAbout'}" >About</router-link> tab.
+          </p>
+        </v-row>
+      </v-container>
+
+      <v-file-input
+        class="d-none"
+        @change="fileInputChange($event)"
+        ref="fileInput"
+        label="File input"
+      ></v-file-input>
+    </div>
+  </div>
+</template>
+
+<script>
+/**
+ * This component handles the UI for adding an Excel file of contexts for the mail merge.
+ * The Excel file is parsed and the contexts within it
+ * are added to the vuex store ('mergeForm.contexts' property)
+ * The contexts are also displayed to the user in a table
+ */
+
+import XLSX from 'xlsx';
+import moment from 'moment';
```

Comment:
As mentioned by Lucas, we should NOT be depending on moment.js if we can help it, especially on new code. Alternatives like date-fns and day.js should be able to do the same things you're looking for.